sklearn/utils/stats.py

```diff
@@ -71,60 +63,38 @@ def _weighted_percentile(array, sample_weight, percentile=50,
     def _squeeze_arr(arr, n_dim):
         return arr[0] if n_dim == 1 else arr
 
-    if interpolation == "nearest":
-        # compute by nearest-rank method
-        # The rank correspond to P / 100 * N; P in [0, 100]
-        # Here, N is replaced by the sum of the weight and P will be adjusted
-        # by the weights
-        adjusted_percentile = percentile * weight_cdf[-1]
-        percentile_value_idx = np.array([
-            np.searchsorted(weight_cdf[:, i], adjusted_percentile[i])
-            for i in range(weight_cdf.shape[1])
+    adjusted_percentile = (weight_cdf - sorted_weights)
```

Comment:
This is the same as removing the last element in `weight_cdf` and adding an element `0` to the start. This might be faster than doing the subtraction?

Reply:
Even if you have some weight to zero?

Reply:
Should be. Easy e.g:
```
x = np.array([2,4,0,4,0,2,0])
np.cumsum(x)
```
array([ 2,  6,  6, 10, 10, 12, 12])


```
np.cumsum(x) - x
```

array([ 0,  2,  6,  6, 10, 10, 12])